tests/framework/validators/ExistValidatorTest.php

```
@@ -295,6 +295,68 @@ public function testSecondTargetAttributeWithError(): void
+        $val = new ExistValidator(['forceMasterDb' => false, 'targetClass' => ValidatorTestRefModel::class, 'targetAttribute' => 'id']);
+        $method = new \ReflectionMethod($val, 'valueExists');
+        $method->setAccessible(true);
+        $this->assertTrue($method->invoke($val, ValidatorTestRefModel::class, ValidatorTestRefModel::find()->andWhere(['id' => 2]), 2));
+    }
+
+    public function testCreateQueryWithFilter(): void
+    {
+        $val = new ExistValidator(['targetClass' => ValidatorTestRefModel::class, 'targetAttribute' => 'id', 'filter' => function ($query) {
+            $query->andWhere(['a_field' => 'ref_to_2']);
+        }]);
+        $method = new \ReflectionMethod($val, 'createQuery');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '352', column '23'). (undefined)

(MissingImport)